assets/js/base/components/cart/quantity-selector/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const QuantitySelector = ( { className, quantity, onChange } ) => {
```

Comment:
I applied similar changes to the `<TextInput>` component in #1523 but after @senadir's review and looking at Gutenberg control components, I'm not 100% sure that's what we want.

As @senadir [said](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1523#discussion_r364723700), setting a default for the `quantity`, the `QuantitySelector` will always be a controlled component. Even if that's the [recommended approach](https://reactjs.org/docs/uncontrolled-components.html) according to React Docs, I'm not sure we want to limit this component to always be used like this in case at some point in the future we allow plugins or other projects to use these components.

Looking at our repo, we have been a bit inconsistent with this. We have some components where we don't set a default value or default onChange (`<SortSelect>`, `<RadioControl>`...), while in some other we do (`<CheckboxList>`, `<DropdownSelector>`...). Should we try to agree on a common solution and always stick to it? Gutenberg doesn't seem to set neither a default value neither a default onChange for its form controls (some examples: [1](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/select-control/index.js#L16), [2](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/range-control/index.js#L18), [3](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/textarea-control/index.js#L11), [4](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/text-control/index.js#L11)).

@haszari I don't think this discussion should block this PR at all, but wanted to bring it here to get your attention and @nerrad's. If you think that's an issue we would like to fix, I would open an issue.

Reply:
I agree we should be consistent and I would like to default to making a low level component like this controlled.  Other use-cases are still possible, they just require a wrapper component that's all. I also think whether it's controlled or not depends on the complexity of the component.

So if we want to be consistent about having low level input type components be controlled, then we should do the following imo:

* don't set a default for the event handlers, but ensure PropTypes defines them as required.
* set default for value.

Regarding the Gutenberg examples you gave, I think they should have a default for the event handler otherwise it's going to error if it's not provided (and they aren't controlled). So I actually wouldn't follow the GB pattern here because imo it's wrong (and if I do work in that component I'd fix it :) )

I agree let's not block this pull here, but let's create an issue to make things consistent for low level input type components. 

Thanks for drawing attention to this Albert!

Reply:
The only drawback I see from this approach is that in `disabled` form controls it's a legit case not to specify an `onChange` prop. But I guess in those cases consumers can just do `onChange={ () => null }` to avoid the warning.

Reply:
> The only drawback I see from this approach is that in disabled form controls it's a legit case not to specify an onChange prop.

I'm sorry I don't follow. `<Disabled />` disables everything on the observed dom in the rendered children. Are you saying that a parent might "disable" an input by not passing through the `onChange` prop?

Reply:
I mean something like this:
```JSX
<TextInput
	...
	disabled={ true }
	onChange={ () => null } // if the input is disabled, we don't really need an `onChange` handle
/>
```
Consumers will be forced to pass a function to `onChange` even in disabled or readonly inputs. Not a big deal but wanted to put it on the table.

Reply:
Oh I see, well in most use-cases I would simply see it being something like this:

```js
<TextInput
	...
	disabled={ isDisabled }
	onChange={ onChange } 
/>
```

...as there'd be some variable condition controlling whether the field is disabled or not. No need to modify any other props when switching the `disabled` prop?

Reply:
Yes, I think in most cases that will not be an issue, so your proposed solution sounds good to me. If the others don't say the opposite, I will create an issue to track implementing it in all form components.

Reply:
Issue created: #1526.